File: go/src/www/js/token.js
Property = {
    key:null,
    value:null,
    add: function(t) {t.extra.push({key:this.key, value:this.value})}
}

Token = function() {
    this.id_token=null
    this.type_token=null
    this.access_token=null
    this.refresh_token=null
    this.expires_in=null
    this.expiry=new Date().toJSON()
    this.extra=[]
    this.status=null
}

Tokeng = function() {
    this.access_token=null
    this.error=null
    this.expires_in=null
    this.state=null
}

File: server go/default/rpc/token.go
package rpc

import (
	"appengine/urlfetch"
	"io/ioutil"
	"encoding/json"
	"net/http"
	"github.com/crhym3/go-endpoints/endpoints"
	"strconv"
	"time"
)

type Property struct {
	Key string `json:"key"`
	Value string `json:"value"`
}

type Token struct {
	Id int64 `json:"id_token"`
	Name string `json:"name_token"`
	Email string `json:"email_token"`
	Type string `json:"type_token"`
	Access string `json:"access_token"`
	Refresh string `json:"refresh_token"`
	ExpiresIn time.Duration `json:"expires_in"`
	Expiry time.Time `json:"expiry"`
	Extra []Property `json:"extra"`
	Status string `json:"status"`
	Message string `json:"message"`
	Client *http.Client `json:"-"`
	Context endpoints.Context `json:"-"`
	Oauth_token string `json:"oauth_token"`
	Oauth_verifier string `json:"oauth_verifier"`
}

func (t *Token) Validate(s String) error {
    error = nill
	return error
}

